src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownValidatorBuilder.cs

```diff
@@ -15,75 +16,202 @@ namespace Microsoft.DocAsCode.Dfm.MarkdownValidators
+            foreach (var setting in settings)
+            {
+                _settings.Add(setting);
+            }
         }
 
-        public void AddValidators(IEnumerable<string> validatorContracts)
+        public void EnsureDefaultValidator()
         {
-            ValidatorContracts = ValidatorContracts.AddRange(validatorContracts);
+            if (!_gobalValidators.ContainsKey(DefaultValidatorName))
+            {
+                _gobalValidators[DefaultValidatorName] = new MarkdownValidationRule
+                {
+                    RuleName = DefaultValidatorName,
+                };
+            }
         }
 
         public IMarkdownTokenRewriter Create()
         {
+            var context = new MarkdownRewriterContext(CompositionHost, GetEnabledTagRules().ToImmutableList());
+            return new MarkdownTokenRewriteWithScope(
+                MarkdownTokenRewriterFactory.FromValidators(
+                    MarkdownValidatePhaseName,
+                    GetEnabledRules().Concat(
+                        new[]
+                        {
+                            MarkdownTokenValidatorFactory.FromLambda<MarkdownTagInlineToken>(context.Validate)
+                        })),
+                MarkdownValidatePhaseName);
+        }
+
+        private IEnumerable<IMarkdownTokenValidator> GetEnabledRules()
+        {
             var list = new List<IMarkdownTokenValidator>();
-            foreach (var contract in ValidatorContracts)
+            HashSet<string> enabledContractName = new HashSet<string>();
+            foreach (var item in _validators)
             {
-                foreach (IMarkdownTokenValidatorProvider vp in CompositionHost.GetExports(typeof(IMarkdownTokenValidatorProvider), contract))
+                if (IsDisabledBySetting(item) ?? item.Rule.Disable)
                 {
-                    list.AddRange(vp.GetValidators());
+                    enabledContractName.Remove(item.Rule.RuleName);
                 }
+                else
+                {
+                    enabledContractName.Add(item.Rule.RuleName);
+                }
+            }
+            foreach (var pair in _gobalValidators)
+            {
+                if (pair.Value.Disable)
+                {
+                    enabledContractName.Remove(pair.Value.RuleName);
+                }
+                else
+                {
+                    enabledContractName.Add(pair.Value.RuleName);
+                }
+            }
+            return from name in enabledContractName
+                   from IMarkdownTokenValidatorProvider vp in CompositionHost.GetExports(typeof(IMarkdownTokenValidatorProvider), name)
+                   from v in vp.GetValidators()
+                   select v;
+        }
+
+        private IEnumerable<MarkdownTagValidationRule> GetEnabledTagRules()
+        {
+            foreach (var item in _tagValidators)
+            {
+                if (IsDisabledBySetting(item) ?? item.Rule.Disable)
```

Comment:
setting is repo level, rule is template level, so repo level override template level.

---
In reply to: [73102156](https://github.com/dotnet/docfx/pull/545#discussion_r73102156) [](ancestors = 73102156)